src/legacy/core_plugins/kibana/public/discover/build_services.ts

```diff
@@ -63,8 +63,10 @@ export interface DiscoverServices {
   getSavedSearchById: (id: string) => Promise<SavedSearch>;
   getSavedSearchUrlById: (id: string) => Promise<string>;
   uiSettings: IUiSettingsClient;
+  search: DataPublicPluginStart['search'];
```

Comment:
You already have `DataPublicPluginStart` on this contract. No need to add search then :)